### PR TITLE
Distinguish controller vs. component actions

### PR DESCRIPTION
**Pull request by hankfanchiu**
In response to: https://github.com/emberjs/guides/issues/1387

- Update autocomplete-component.md.
- Add distinction between the `filter-listing` component's actions and the `index` controller's actions,  by prefixing the controller's actions with `controller`- to reduce the chance of any confusion.
- Add safety check for `filter` to be not `null` nor empty string, before querying.
- Set `model` to the query result's `data` property (adhering to the JSON:API)

### CODE REVIEW

**Comment by locks** [comment]
@hankfanchiu could you separate the action renaming from the rest please? It would make it easier to review.

**Comment by toddjordan** [comment]
Hey @hankfanchiu thanks for the pr.

I'm against referring to a controller from a component.  I don't think our filter-listing component should care whether its forwarding its action to a controller, or just a parent controller, or even a route (using the route helper addon).

What I would prefer to see is the filter-listing not even define actions and just forward its actions to its parent.  You can do this by using the action helper with an unquoted argument in the filter-listing template.  That way there is only one thing handling the action, so there is not need to make a distinction for clarity.

File: source/localizable/tutorial/autocomplete-component.md [hunk 1]
```diff
@@ -15,7 +15,8 @@ As before, this creates a Handlebars template
 The Handlebars template looks like this:
 
 ```app/templates/components/filter-listing.hbs
-City: {{input value=filter key-up=(action 'autoComplete')}} 
+City: {{input value=filter key-up=(action 'autoComplete')}}
```
Comment (toddjordan) [inline source/localizable/tutorial/autocomplete-component.md]: Need to try this but I think you can just call it like this and remove the need for handling the action within the component.

```hbs
{{input value=filter key-up=(action autoComplete filter)}}
```

Comment (hankfanchiu) [inline source/localizable/tutorial/autocomplete-component.md]: How about this change in the actual app? https://github.com/emberjs/super-rentals/pull/17

If that's good, then I'll update this PR for the guide to reflect the app.

**Comment by toddjordan** [comment]
Thanks @hankfanchiu looks good.  go ahead and update the guides

**Comment by toddjordan** [comment]
Looks great @hankfanchiu, thanks for contributing!

**Comment by toddjordan** [comment]
@homu r+

**Pull request merged**